amplipi/display/einkdisplay.py

```diff
@@ -150,17 +196,159 @@ def update_display(self, info: SysInfo = SysInfo(None, None, None), draw_base=Fa
     try:
-      for i in range(20, 10, -1):
+      for i in range(max_size, min_size, -1):
         font = ImageFont.truetype(self.fontname, i)
-        if font.getlength(password) <= max_length:
+        if font.getlength("Pass: " + password) <= max_length:
           return font
     except Exception as exc:
       raise Exception(f'Failed to load {self.fontname} font') from exc
     return ImageFont.truetype(self.fontname, 10)
 
 
-def get_info(iface, default_pass) -> SysInfo:
+def update_logs(status) -> None:
+  """Update the status log files by appending the current status and time"""
+  data = {}
+  try:
+    with open("display-logs.json", "r") as log_file:
+      txt = log_file.read()
+      data = json.loads(txt)
+  except Exception as e:
+    log.exception(
+      f"Could not read eink display logs: {e}\n Proceeding as if it's empty. This might occur on first run.")
+
+  try:
+    data[str(datetime.datetime.now())] = status
+    with open("display-logs.json", "w+") as log_file:
+      log_file.write(json.dumps(data))
+  except Exception as e:
+    log.exception(f"Error writing to log file: {e}")
+
+
+def load_status_from_file() -> Optional[Union[str, int]]:
+  """Load a status from the status file"""
+  try:
+    with open(status_filename, "r") as f:
+      data = f.read().split(",")
+      if len(data) == 0:
+        return None
+      # Ignore the status if it is past its expiration date
+      if data[1] != "None":
+        expiration_date = datetime.datetime.strptime(data[1], '%Y-%m-%d %H:%M:%S.%f')
+        if expiration_date < datetime.datetime.now():
+          return None
+
+      if data[0] == "None":
+        return None
+      if data[0].isdigit():
+        return int(data[0])
+      return data[0]
+  except FileNotFoundError:
+    return None
+  except Exception as e:
+    log.exception(f"Failed to load status: {e}")
+  return 14
+
+
+def get_num_expanders(url: str) -> int:
+  try:
+    req = requests.get(url, timeout=0.2)
+    if req.status_code == 200:
+      status = models.Status(**req.json())
+      info = status.info
+      if info is not None:
+        return len(info.expanders)
+      else:
+        set_custom_eink_status(17, None)
+        return 0
+    else:
+      set_custom_eink_status(17, None)
+      return 0
+  except Exception as e:
+    log.exception(f'Exception getting number of expanders: {e}')
+    set_custom_eink_status(16, None)
+    return 0
+
+
+def get_status(url: str) -> Tuple[Union[str, int], Optional[int], int]:
+  """Returns the system's status code as either a str or int, a serial number, and how many expanders are hooked up"""
+
+  serial = -1
+  result_status: Union[str, int] = 0
+  st = load_status_from_file()
+  if st == startup_msg:
+    update_logs(st)
+    result_status = st
+
+  # Check if API is running
+  api_on = os.system("systemctl --user is-active amplipi.service")
+  if api_on != 0:
+    return 0, None, 0
+
+  if url is None:
+    return 10, None, 0
```

Comment:
All of these exception codes should be an enum. For an example see `eeprom.py`'s `class UnitType(IntEnum)`.